### ISSUE TITLE
MissingPackageDeclaration - Incorrect issue reported.

### ISSUE THREAD

**Issue by Abhimanyu14** [detekt/detekt#7296]
MissingPackageDeclaration rule is reporting an issue for a file that does have package declaration.

## Expected Behavior
No issues

## Observed Behavior
MissingPackageDeclaration rule is reporting an issue for a file that does have package declaration.

## Steps to Reproduce
Running detekt checks.

## Your Environment

* Version of detekt used:
* Version of Gradle used (if applicable):
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version:
* Link to your project (if it's a public repository): https://github.com/Abhimanyu14/finance-manager

Note: 
Issue reported only for a particular file. 

> /Users/abhimanyu/Documents/projects/finance-manager/codebase/android/core/common/src/main/java/com/makeappssimple/abhimanyu/financemanager/android/core/common/extensions/StringExtension.kt:1:1: The file does not contain a package declaration. [MissingPackageDeclaration]

```
package com.makeappssimple.abhimanyu.financemanager.android.core.common.extensions

import kotlin.contracts.contract

public fun String?.isNotNullOrBlank(): Boolean {
    contract {
        returns(true) implies (this@isNotNullOrBlank != null)
    }
    return !this.isNullOrBlank()
}

// ...
```

**Comment by atulgpt** [comment]
Hi @Abhimanyu14 that is strage running the UT 

```
@Test
    fun `sshould report if package declaration is missing`() {
        val code = """
            package com.makeappssimple.abhimanyu.financemanager.android.core.common.extensions

            import kotlin.contracts.contract

            public fun String?.isNotNullOrBlank(): Boolean {
                contract {
                    returns(true) implies (this@isNotNullOrBlank != null)
                }
                return !this.isNullOrBlank()
            }
        """.trimIndent()

        val findings = MissingPackageDeclaration(Config.empty).compileAndLint(code)

        assertThat(findings).isEmpty()
    }
```

passes on the latest develop. Is this compiler generated file?
